### DISCUSSION TITLE
Using with a Chrome Extension (Manifest V3)?

### DISCUSSION THREAD

**Discussion by ShaneOH** [floating-ui/floating-ui#2369] [Q&A]
I'm trying to make pretty tooltips in my chrome extension and I'm having a hard time getting both this and `tippyjs` to work with Manifest V3. Manifest V3 does not allow remote code and recommends [bundling the files instead](https://developer.chrome.com/docs/extensions/migrating/improve-security/#remotely-hosted-code).

For Floating UI, I tried the following:

```
curl https://cdn.jsdelivr.net/npm/@floating-ui/core@1.2.6 > floating-ui-core.js
curl https://cdn.jsdelivr.net/npm/@floating-ui/dom@1.2.9 > floating-ui-dom.js  
```

`manifest.json`
```
{
   "manifest_version": 3,
   "name": "...",
   "version": "...",
   "content_scripts":
   [
      {
         "matches": ["*://*/*"],
         "js": [
            "./floating-ui-dom.js",
            "./floating-ui-core.js",
            "./main.js"
         ],
         "css" : ["styles.css"],
         "run_at": "document_end",
         "type": "module"
      }
   ]
}
```

`main.js`
```
    window.FloatingUIDOM.computePosition(element, tooltip).then(({x, y}) => {
      Object.assign(tooltip.style, {
        left: `${x}px`,
        top: `${y}px`,
      });
    });
```

All this just ends up with the following:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'computePosition')
```

I tried using a background / service worker too, but couldn't get that working either.

Is there a guide somewhere on how to use this library in a Chrome Extension?

**Comment by FezVrasta** [comment]
Maybe try writing a small script that sets window.test = true and see if that is available from main.js? At least you make sure the setup is correct.

**Comment by ShaneOH** [comment]
Hey @FezVrasta I did try that and saw it was true (for a content_script test and not for the background test). Tried some other stuff that didn't work. Then tried to back to the original setup to try again and just happened to write the files in my `manifest.json` in a different order:

```
         "js": [
            "./floating-ui-core.js",
            "./floating-ui-dom.js",
            "./main.js"
         ],
```

And that seems to have fixed the original issue. 

So if anyone has any guides or examples for this or tippyjs for Chrome Extension development that would be much appreciated as it's quite a bit more difficult to work with than node thus far since I'm newer to it, but for now I'm unblocked at least 😅
